doc/generic_callback_wrapper.md

```diff
@@ -0,0 +1,77 @@
+# GenericCallbackWrapper
+
+`Napi::GenericCallbackWrapper<T>` is a class with one template parameter `T`, 
+which is intended to allow you to return `T` from a C++ function running arbitrary async i.e. concurrency 
+regimes like `boost::asio` via a `Napi::Promise` i.e. JS promise, including exception and conversion function handling. 
+
+`Napi::GenericCallbackWrapper<T>` requires C++ exceptions enabled, so works by default.
+
+All you need is the `Napi::Promise::Deferred` you want to Resolve/Reject, a function to convert your type `T` 
+to a `Napi::Value` to construct your wrapper, which then can give you a `std::function` callback, which
+can be called later from any custom thread, in order to resolve/reject the promise. 
+Life time handling included. That means that the wrapper is internally reference counted and don't need to be stored, 
+the function object you get from `get_native_callback` will ensure the lifetime.
```

Comment:
It would be useful to document the methods/parameters on GenericCallbackWrapper that can be called by code that uses the class. That would include the constructor, set_unsubscription_function, and get_native_callback.  That would be consistent with the docs for other parts of node-addon-api I think help explain how it is used.

Same comment for the GenericSubscriptionWrapper as well.